### ISSUE TITLE
Dataframe Algebra

### ISSUE THREAD

**Issue by foudrer** [modin-project/modin#3035]
Hi, the paper said, there are 15 basic API of modin which can express many pandas dataframe api. I read the code of modin, coufused by the 15 basic API. Do yuo have any suggestions?

**Comment by devin-petersohn** [comment]
@foudrer Yes, there is a mix of public and private APIs in the BasePandasFrame API. We have a couple of efforts to clean it up via #3030 and #2744.
